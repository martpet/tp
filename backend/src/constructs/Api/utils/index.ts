export * from './cookieName';
export * from './errorResponse';
export * from './getIdTokenPayload';
export * from './lambdaEdgeErrorResponse';
export * from './parseEventCookies';
export * from './parseLambdaEdgeEventCookies';
export * from './testHelpers';
