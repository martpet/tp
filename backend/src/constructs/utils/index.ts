export * from './createEdgeFunction';
export * from './createNodejsFunction';
