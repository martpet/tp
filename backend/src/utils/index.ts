export * from '~/../../shared/utils';
//
export * from './constructsUtils';
export * from './createTableOptions';
export * from './getEnvName';
export * from './getRandomBase64UrlSafe';
export * from './getRoleCredentials';
