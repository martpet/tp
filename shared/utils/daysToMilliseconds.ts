export const daysToMilliseconds = (days: number) => days * 24 * 60 * 60 * 1000;
