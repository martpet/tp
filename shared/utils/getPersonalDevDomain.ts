import { rootDomain } from '../consts/sharedConsts';

const subdomainKey = 'VITE_PERSONAL_SUBDOMAIN';

export const getPersonalDevDomain = (env: any) => {
  // @ts-ignore
  const isBrowser = typeof window !== 'undefined';

  if (
    (isBrowser && !env.DEV) ||
    (!isBrowser && !['development', 'test'].includes(env.NODE_ENV))
  ) {
    return '';
  }

  const subdomain = env[subdomainKey];

  if (!subdomain) {
    throw Error(`"${subdomainKey}" missing in .env.local`);
  }

  const sanitizedSubdomain = subdomain.replace(/_|-$|\.|\*|\\|\//gi, '');

  return `${sanitizedSubdomain}.dev.${rootDomain}`;
};
