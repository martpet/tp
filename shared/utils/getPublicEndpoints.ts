import { ApiMethod, ApiPath, ApiRoutes } from '../types';

export const getPublicEndpoints = (apiRoutes: ApiRoutes) => {
  const result: Partial<Record<ApiPath, ApiMethod[]>> = {};

  Object.entries(apiRoutes as ApiRoutes).forEach(([path, { methods }]) => {
    const publicMethods: ApiMethod[] = [];

    Object.entries(methods).forEach(([method, { isPublic }]) => {
      if (isPublic) {
        publicMethods.push(method as ApiMethod);
      }
    });

    if (publicMethods.length) {
      result[path as ApiPath] = publicMethods;
    }
  });

  return result;
};
