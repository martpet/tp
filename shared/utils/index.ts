export * from './capitalize';
export * from './daysToMilliseconds';
export * from './filterChangedProps';
export * from './getPersonalDevDomain';
