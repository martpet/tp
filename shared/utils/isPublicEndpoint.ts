import { publicEndpoints } from '../consts/publicEndpoints';
import { ApiMethod, ApiPath } from '../types';

type Props = {
  path: string;
  method: string;
};

export function isPublicEndpoint({ path, method }: Props) {
  return Boolean(publicEndpoints[path as ApiPath]?.includes(method as ApiMethod));
}
