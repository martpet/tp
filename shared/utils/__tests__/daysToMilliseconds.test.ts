import { daysToMilliseconds } from '~/utils';

describe('daysToMilliseconds', () => {
  it('returns a correct value', () => {
    expect(daysToMilliseconds(5)).toMatchSnapshot();
  });
});
