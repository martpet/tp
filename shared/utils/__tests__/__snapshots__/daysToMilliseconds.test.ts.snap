// Vitest Snapshot v1

exports[`daysToMilliseconds > returns a correct value 1`] = `432000000`;
